enterprise/server/testutil/testredis/testredis.go

```diff
@@ -83,27 +83,34 @@ func (h *Handle) start() {
 	if err != nil {
 		assert.FailNowf(h.t, "redis binary could not be started", err.Error())
 	}
+	h.t.Cleanup(func() {
+		h.Shutdown()
+	})
 	go func() {
-		if err := cmd.Wait(); err != nil {
-			close(h.done)
-			select {
-			case <-ctx.Done():
-			default:
-				log.Warningf("redis server did not exit cleanly: %v", err)
-			}
+		err := cmd.Wait()
+		close(h.done)
+		if ctx.Err() != nil {
+			// Context cancelled; test was cleaned up normally.
+			return
 		}
+		h.t.Fatalf("redis server exited unexpectedly: %s", err)
```

Comment:
Flip the conditional above and put this line inside?

Reply:
Done